--- a/v2h/drivers/drpai/drpai-if.c	2026-09-17 13:01:39.626278600 +0700
+++ b/v2h/drivers/drpai/drpai-if.c	2026-09-17 13:01:48.972786500 +0700
@@ -35,6 +35,7 @@
 #include <linux/poll.h>
 #include <linux/sched.h>
 #include <linux/slab.h>
+#include <linux/vmalloc.h>
 #include <linux/types.h>
 #include <linux/uaccess.h>
 #include <linux/wait.h>
@@ -168,7 +169,7 @@
 
 /* A function called from the kernel */
 static int drpai_probe(struct platform_device *pdev);
-static int drpai_remove(struct platform_device *pdev);
+static void drpai_remove(struct platform_device *pdev);
 static int drpai_open(struct inode *inode, struct file *file);
 #ifdef ENABLE_DRP_SUPPORT_SHARED_MEMORY
 static void drpai_shutdown(struct platform_device *pdev);
@@ -276,7 +277,7 @@
     uint32_t drpai_divfix;
 };
 
-static DEFINE_SEMAPHORE(rw_sem);
+static DEFINE_SEMAPHORE(rw_sem, 1);
 static struct drpai_rw_status *drpai_rw_sentinel;
 
 /* Virtual base address of register */
@@ -350,7 +351,7 @@
     return ret;
 }
 
-static int drpai_remove(struct platform_device *pdev)
+static void drpai_remove(struct platform_device *pdev)
 {
     DRPAI_DEBUG_PRINT("start.\n");
 
@@ -359,7 +360,6 @@
 
     DRPAI_DEBUG_PRINT("end.\n");
 
-    return 0;
 }
 
 #ifdef ENABLE_DRP_SUPPORT_SHARED_MEMORY
@@ -1283,7 +1283,7 @@
     }
 
     /* Cleate class "/sys/class/drpai/" */
-    drpai_class = class_create(THIS_MODULE, DRPAI_DRIVER_NAME);
+    drpai_class = class_create(DRPAI_DRIVER_NAME);
     if (IS_ERR(drpai_class)) {
         ptr_err = PTR_ERR(drpai_class);
         pr_err("DRP-AI Driver: class_create = %d\n", ptr_err);
