--- a/uvcs_kernel_package/src/lkm/uvcs_lkm.c	2026-09-17 12:44:50.728392400 +0700
+++ b/uvcs_kernel_package/src/lkm/uvcs_lkm.c	2026-09-17 12:45:20.168686200 +0700
@@ -861,12 +861,12 @@
 	return result;
 }
 
-static int uvcs_dev_remove(struct platform_device *pdev)
+static void uvcs_dev_remove(struct platform_device *pdev)
 {
 	struct uvcs_platform_data *pdata = pdev->dev.platform_data;
 
 	if (!pdata)
-		return -EFAULT;
+		return;
 
 	if (pdata->device_vcp)
 		uvcs_put_vcp_resource((struct uvcs_vcp_hwinf *)pdata->hwinf);
@@ -877,8 +877,6 @@
 	uvcsdrv->pdev[pdata->device_id] = NULL;
 	platform_set_drvdata(pdev, NULL);
 	vfree(pdata);
-
-	return 0;
 }
 
 /******************************************************************************/
