--- a/uvcs_kernel_package/src/lkm/uvcs_lkm_uf_io.c	2026-09-17 12:46:52.114741700 +0700
+++ b/uvcs_kernel_package/src/lkm/uvcs_lkm_uf_io.c	2026-09-17 12:46:52.269463400 +0700
@@ -197,7 +197,7 @@
 			if ((driver_info->vcpinf[i].irq_vlc == irq)
 			&&	(driver_info->vcpinf[i].pdev == pdev)) {
 				spin_lock_irqsave(&driver_info->vcpinf[i].slock_vlc, flags);
-				del_timer(&driver_info->vcpinf[i].timer_vlc);
+				timer_delete(&driver_info->vcpinf[i].timer_vlc);
 				iowrite32(0u, driver_info->vcpinf[i].reg_vlc + UVCS_VCPREG_IRQENB);
 				uvcs_cmn_vlc_interrupt(driver_info->uvcs_info, i, ts.tv_nsec, UVCS_FALSE);
 				spin_unlock_irqrestore(&driver_info->vcpinf[i].slock_vlc, flags);
@@ -231,7 +231,7 @@
 			if ((driver_info->vcpinf[i].irq_ce == irq)
 			&&	(driver_info->vcpinf[i].pdev == pdev)) {
 				spin_lock_irqsave(&driver_info->vcpinf[i].slock_ce, flags);
-				del_timer(&driver_info->vcpinf[i].timer_ce);
+				timer_delete(&driver_info->vcpinf[i].timer_ce);
 				iowrite32(0u, driver_info->vcpinf[i].reg_ce + UVCS_VCPREG_IRQENB);
 				uvcs_cmn_ce_interrupt(driver_info->uvcs_info, i, ts.tv_nsec, UVCS_FALSE);
 				spin_unlock_irqrestore(&driver_info->vcpinf[i].slock_ce, flags);
@@ -250,7 +250,7 @@
 	struct timer_list *data_ptr /**< [in] hardware identifier */
 	)
 {
-	struct uvcs_vcp_hwinf *vcpinfo = from_timer(vcpinfo, data_ptr, timer_vlc);
+	struct uvcs_vcp_hwinf *vcpinfo = timer_container_of(vcpinfo, data_ptr, timer_vlc);
 	u64 data = vcpinfo -> timer_vlc_data;
 	if (driver_info) {
 		struct timespec64 ts = {0u};
@@ -259,7 +259,7 @@
 		ktime_get_raw_ts64(&ts);
 		if (data < driver_info->vcp_devnum) {
 			spin_lock_irqsave(&driver_info->vcpinf[data].slock_vlc, flags);
-			del_timer(&driver_info->vcpinf[data].timer_vlc);
+			timer_delete(&driver_info->vcpinf[data].timer_vlc);
 			iowrite32(0u, driver_info->vcpinf[data].reg_vlc + UVCS_VCPREG_IRQENB);
 			uvcs_cmn_vlc_interrupt(driver_info->uvcs_info, data, ts.tv_nsec, UVCS_TRUE);
 			spin_unlock_irqrestore(&driver_info->vcpinf[data].slock_vlc, flags);
@@ -274,7 +274,7 @@
 	struct timer_list *data_ptr /**< [in] hardware identifier */
 	)
 {
-	struct uvcs_vcp_hwinf *vcpinfo = from_timer(vcpinfo, data_ptr, timer_ce);
+	struct uvcs_vcp_hwinf *vcpinfo = timer_container_of(vcpinfo, data_ptr, timer_ce);
 	u64 data = vcpinfo -> timer_ce_data;
 	if (driver_info) {
 		struct timespec64 ts = {0u};
@@ -283,7 +283,7 @@
 		ktime_get_raw_ts64(&ts);
 		if (data < driver_info->vcp_devnum) {
 			spin_lock_irqsave(&driver_info->vcpinf[data].slock_ce, flags);
-			del_timer(&driver_info->vcpinf[data].timer_ce);
+			timer_delete(&driver_info->vcpinf[data].timer_ce);
 			iowrite32(0u, driver_info->vcpinf[data].reg_ce + UVCS_VCPREG_IRQENB);
 			uvcs_cmn_ce_interrupt(driver_info->uvcs_info, data, ts.tv_nsec, UVCS_TRUE);
 			spin_unlock_irqrestore(&driver_info->vcpinf[data].slock_ce, flags);
@@ -979,7 +979,7 @@
 	snprintf(vcpinf->irq_name_ce, sizeof(vcpinf->irq_name_ce), "%s %s,c",
 		 dev_name(&pdev->dev), name);
 
-	rst = devm_reset_control_array_get(&pdev->dev, false, false);
+	rst = devm_reset_control_array_get_exclusive(&pdev->dev);
 	if (IS_ERR(rst))
 	{
 		dev_err(&pdev->dev, "failed to get reset array\n");
